contrib/endpoints/src/api_manager/method_impl.cc

```diff
@@ -51,17 +50,17 @@ void MethodInfoImpl::addAudiencesForIssuer(const string &issuer,
   if (iss.empty()) {
     return;
   }
+
   set<string> &audiences = issuer_audiences_map_[iss];
-  stringstream ss(audiences_list);
-  string audience;
-  // Audience list is comma-delimited.
-  while (getline(ss, audience, ',')) {
-    if (!audience.empty()) {  // Only adds non-empty audience.
-      std::string aud = utils::GetUrlContent(audience);
-      if (!aud.empty()) {
-        audiences.insert(aud);
-      }
+
+  // Audience list is comma-delimited with possible white spaces.
+  char *tokens = strtok(const_cast<char *>(audiences_list.c_str()), " ,");
```

Comment:
this still dangerous.  std::string will not expect anyone modifying its buffer.  Even you have your own copy of string,  internally, it could still use ref_count, since it is not expecting you to modify the buffer.

You can create a
char buf[str.size() + 1);

use that.